src/librustc/sir.rs

```diff
@@ -102,7 +104,79 @@ impl SirFuncCx {
 
     /// Converts a MIR statement to SIR, appending the result to `bb`.
     pub fn codegen_statement(&mut self, bb: ykpack::BasicBlockIndex, stmt: &mir::Statement<'_>) {
-        // FIXME: Nothing is implemented yet.
-        self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)));
+        match stmt.kind {
+            mir::StatementKind::Assign(box (ref place, ref rvalue)) => {
+                let assign = self.lower_assign_stmt(place, rvalue);
+                self.push_stmt(bb, assign);
+            },
+            _ => self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)))
+        }
+    }
+
+    fn lower_assign_stmt(&self, lvalue: &mir::Place<'_>, rvalue: &mir::Rvalue<'_>) -> ykpack::Statement {
+        let lhs = self.lower_local(&lvalue.local);
```

Comment:
Please either use a new function `lower_place`, or return `Statement::Unimplemented` when there is a projection. This kind of will implement later makes it hard to debug miscompilations later on. (I speak from experience)

Reply:
I made a note so I won't forget about `projection` later, but adding `Unimplemented` now is probably a better idea. Fixed in 8ba020b